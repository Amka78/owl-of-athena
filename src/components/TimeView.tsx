//#region Import Modules
import React, { FunctionComponent } from "react";
import { Text, StyleSheet, TextStyle, View } from "react-native";
import { Colors, Fonts } from "../constants";
export type TimeViewMode = "meridian" | "time";
//#endregion

//#region Types
export type TimeViewProps = {
    timeViewStyle?: TextStyle;
    timeStyle?: TextStyle;
    timeMeridianStyle?: TextStyle;
    hours: number;
    minutes: number;
    mode: TimeViewMode;
};
//#endregion

//#region Component
export const TimeView: FunctionComponent<TimeViewProps> = (
    props: TimeViewProps
) => {
    return (
        <View style={[style.alarmView, props.timeViewStyle]}>
            <Text style={[style.alarmTime, props.timeStyle]}>
                {getTimeText(props)}
            </Text>
            {props.mode === "meridian" ? (
                <Text style={[style.alarmMeridian, props.timeMeridianStyle]}>
                    {props.hours > 12 ? "pm" : "am"}
                </Text>
            ) : undefined}
        </View>
    );
};
//#endregion

//#region Styles
const style = StyleSheet.create({
    alarmView: {
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        flex: 1,
    },
    alarmTime: {
        color: Colors.cyan,
        fontFamily: Fonts.primarySemiBold,
        fontSize: 20,
    },
    alarmMeridian: {
        color: Colors.cyan,
        fontFamily: Fonts.primaryRegular,
        fontSize: 20,
    },
});
//#endregion

function getTimeText(props: TimeViewProps): string {
    let timeText: string;

    if (props.mode === "meridian") {
        timeText = `${props.hours > 12 ? props.hours - 12 : props.hours}:${(
            "00" + props.minutes
        ).slice(-2)}`;
    } else {
        timeText = `${props.hours ? props.hours + "h" : ""} ${props.minutes}m`;
    }
    return timeText;
}
