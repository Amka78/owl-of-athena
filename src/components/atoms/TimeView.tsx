//#region Import Modules
import React, { FunctionComponent } from "react";
import { StyleSheet, Text, TextStyle, View } from "react-native";
import { useTheme } from "react-native-paper";

import { Fonts } from "../../constants";
//#endregion

//#region Types
export type TimeViewMode = "meridian" | "time";
export type TimeViewProps = {
    timeViewStyle?: TextStyle;
    timeStyle?: TextStyle;
    timeMeridianStyle?: TextStyle;
    hours: number;
    minutes: number;
    mode: TimeViewMode;
};
//#endregion

//#region Component
export const TimeView: FunctionComponent<TimeViewProps> = (
    props: TimeViewProps
) => {
    const theme = useTheme();
    return (
        <View style={[style.alarmView, props.timeViewStyle]}>
            <Text
                style={[
                    style.alarmTime,
                    { color: theme.colors?.accent },
                    props.timeStyle,
                ]}
            >
                {getTimeText(props)}
            </Text>
            {props.mode === "meridian" ? (
                <Text
                    style={[
                        style.alarmMeridian,
                        { color: theme.colors?.accent },
                        props.timeMeridianStyle,
                    ]}
                >
                    {props.hours > 12 ? "pm" : "am"}
                </Text>
            ) : undefined}
        </View>
    );
};
//#endregion

//#region Styles
const style = StyleSheet.create({
    alarmView: {
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
    },
    alarmTime: {
        fontFamily: Fonts.primarySemiBold,
        fontSize: 20,
    },
    alarmMeridian: {
        fontFamily: Fonts.primaryRegular,
        fontSize: 20,
    },
});
//#endregion

function getTimeText(props: TimeViewProps): string {
    let timeText: string;

    if (props.mode === "meridian") {
        timeText = `${props.hours > 12 ? props.hours - 12 : props.hours}:${(
            "00" + props.minutes
        ).slice(-2)}`;
    } else {
        timeText = `${props.hours ? props.hours + "h" : ""} ${props.minutes}m`;
    }
    return timeText;
}
