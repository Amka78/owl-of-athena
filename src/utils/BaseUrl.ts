import Constants from "expo-constants";

/**
 * Manage API-URL for each environment.
 *
 * @export
 * @class BaseUrl
 */
export default class BaseUrl {
    public static get(relCh?: string) {
        if (!relCh && Constants.manifest.releaseChannel) {
            relCh = Constants.manifest.releaseChannel;
        }

        if (relCh !== undefined) {
            if (relCh!.includes("staging")) {
                return IWinksRestAPI.staging.url;
            } else if (relCh!.includes("prod")) {
                return IWinksRestAPI.prod.url;
            }
        }

        return IWinksRestAPI.dev.url;
    }

    public static getDownloadUrl(
        image: string | string[],
        index?: number
    ): string | undefined {
        if (typeof image === "undefined") {
            return undefined;
        } else if (typeof image === "string") {
            return BaseUrl.get() + image;
        } else {
            if (index === undefined) {
                index = 0;
            }
            return BaseUrl.get() + image![index];
        }
    }
}

export const IWinksRestAPI = {
    dev: {
        url: "https://api.iwinks.io/"
    },
    staging: {
        url: "https://api.iwinks.io/"
    },
    prod: {
        url: "https://api.iwinks.io/"
    }
};
