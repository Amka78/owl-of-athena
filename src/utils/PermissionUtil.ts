import * as Permissions from "expo-permissions";
import { Linking } from "react-native";

export default class PermissionUtil {
    public static async getPermission(permission: any) {
        const { status } = await Permissions.askAsync(permission);
        if (status !== "granted") {
            Linking.openURL("app-settings:");
            return false;
        }
        return true;
    }
}
