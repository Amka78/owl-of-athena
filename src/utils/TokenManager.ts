import { Platform } from "@unimodules/core";
import * as SecureStore from "expo-secure-store";
import { AsyncStorage } from "react-native";

enum SecureStoreKeys {
    JWT_TOKEN = "jwt_token"
}
/**
 * Token Management Class
 *
 * @export
 * @class TokenManager
 */
export default class TokenManager {
    public static async hasToken(): Promise<boolean> {
        const result = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        return result !== undefined;
    }

    /**
     * get token
     *
     * @static
     * @returns {(Promise<string | undefined>)}
     * @memberof TokenManager
     */
    public static async get(): Promise<string | undefined> {
        const token = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        if (token) {
            return token;
        }

        return undefined;
    }

    /**
     * set token
     *
     * @static
     * @param {TokenDto} value
     * @memberof TokenManager
     */
    public static async set(value: string) {
        await this.setPerPlatform(SecureStoreKeys.JWT_TOKEN, value);
    }

    public static async reset() {
        await this.deletePerPlatform(SecureStoreKeys.JWT_TOKEN);
    }

    private static async getPerPlatform(key: string): Promise<string | null> {
        if (Platform.OS === "web") {
            return await AsyncStorage.getItem(key);
        }

        return await SecureStore.getItemAsync(key);
    }

    private static async setPerPlatform(
        key: string,
        value: string
    ): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.setItem(key, value);
        }

        return await SecureStore.setItemAsync(key, value);
    }

    private static async deletePerPlatform(key: string): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.removeItem(key);
        }

        return await SecureStore.deleteItemAsync(key);
    }
}
