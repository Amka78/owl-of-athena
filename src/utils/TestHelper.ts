import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
