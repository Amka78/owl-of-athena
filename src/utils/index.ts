import BaseUrl from "./BaseUrl";
export { BaseUrl };
