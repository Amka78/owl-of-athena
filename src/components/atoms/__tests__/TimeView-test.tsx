//#region Import Modules
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { createMock, toJson } from "../../../utils/TestHelper";
import { TimeView, TimeViewProps } from "../TimeView";
//#endregion

//#region Test
let component: ShallowWrapper<TimeViewProps, unknown, unknown>;
describe("TimeView UnitTest", () => {
    it("renders correctly", () => {
        component = createMock(
            <TimeView hours={12} minutes={50} mode={"meridian"}></TimeView>
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
//#endregion
