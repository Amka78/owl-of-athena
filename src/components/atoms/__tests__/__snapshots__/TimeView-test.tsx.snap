// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeView UnitTest renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "calibre_app_semibold",
          "fontSize": 20,
        },
        Object {
          "color": "#03dac4",
        },
        undefined,
      ]
    }
  >
    12:50
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "calibre_app_regular",
          "fontSize": 20,
        },
        Object {
          "color": "#03dac4",
        },
        undefined,
      ]
    }
  >
    am
  </Text>
</View>
`;
